server/lease/lease.go

```diff
@@ -104,8 +104,8 @@ func (l *Lease) Demoted() bool {
 
 // Keys returns all the keys attached to the lease.
 func (l *Lease) Keys() []string {
-	l.mu.RLock()
 	keys := make([]string, 0, len(l.itemSet))
+	l.mu.RLock()
```

Comment:
This seems not correct. `len(l.itemSet)` also needs to be protected by the lock, since it accesses the `l`